test/app/components/conditional_render_component.rb

```diff
@@ -1,11 +1,11 @@
 # frozen_string_literal: true
 
 class ConditionalRenderComponent < ActionView::Component::Base
-  def initialize(should_render:)
-    @should_render = should_render
+  before_render do
+    throw(:abort) unless @should_render
```

Comment:
Can we use a terminator so that returning `false` halts execution? That feels more natural that having to `throw`

Reply:
ActiveSupport::Callbacks use `throw` terminator by default. We can change to `true`/`false` resolvation, but i`m not sure it`s good idea. It this case you have to control return value of `before_render`

Reply:
controlling for the return value would be fine in most cases. I wonder what the alternatives are (throwing exceptions for control flow is an anti pattern most of the time). In Rails Controllers a before filter that renders or redirects results in the filter chain halting. Can we isolate some set of method calls that would have the same effect?

Reply:
This set may include something like `stop_render` and I can't suggest nothing more. I think it would be even worse than `throw` because it is additional DSL user need to know.

Reply:
This is not principal for me actually. The decision is on your side